src/tools/rbd_mirror/image_map/ImageAction.h

```diff
@@ -0,0 +1,122 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_IMAGE_MAP_IMAGE_ACTION_H
+#define CEPH_RBD_MIRROR_IMAGE_MAP_IMAGE_ACTION_H
+
+#include <map>
+#include <list>
+#include <string>
+
+#include "Types.h"
+#include "common/Mutex.h"
+
+namespace rbd {
+namespace mirror {
+namespace image_map {
+
+class ImageAction {
+public:
+  ImageAction();
+
+  enum ActionType {
+    ACTION_TYPE_ADD = 0,
+    ACTION_TYPE_REMOVE,
+    ACTION_TYPE_SHUFFLE,
+  };
+
+  enum SubActionType {
+    SUB_ACTION_TYPE_INVALID = 0,
```

Comment:
Nit: can we drop this and just use ```COMPLETE```?

Reply:
yeh ok -- this will go away with the `assert` during invalid transition.